src/Microsoft.AspNetCore.Sockets/Transports/ServerSentEventsTransport.cs

```diff
@@ -37,6 +37,9 @@ public ServerSentEventsTransport(ReadableChannel<Message> application, ILoggerFa
 
             context.Response.Headers["Content-Encoding"] = "identity";
 
+            // Workaround for a Firefox bug where EventSource won't fire the open event
+            // until it receives some data
+            await context.Response.WriteAsync(":\r\n");
```

Comment:
We could add handling for an init data message. We would just add some additional logic in the GetMessageType logic [here](https://github.com/aspnet/SignalR/blob/dev/src/Microsoft.AspNetCore.Sockets.Common/Internal/Formatters/ServerSentEventsMessageParser.cs#L180) to check for the init message